File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/CartItemRelationship.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "cart_item_relationships", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "cart_item_relationships_unique", columnNames = {"cart_id", "item_id"})
})
public class CartItemRelationship {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cart_item_relationship_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "cart_id")
    private Cart cart;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "item_id")
    private Item item;

    @NotNull
    @Column(name = "quantity")
    private Long quantity;

}

File: umbrella-insurance-react-ui/src/models/cartItemRelationships/v1/CartItemRelationship.ts
import { Cart } from "../../carts/v1/Cart";
import { Item } from "../../items/v1/Item";

export class CartItemRelationship {
    id?: number;
    cart?: Cart;
    item?: Item;
    quantity?: number;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.cart) {
                this.cart = new Cart(obj.cart);
            }
            if(obj?.item) {
                this.item = new Item(obj.item);
            }
            if(obj?.quantity) {
                this.quantity = obj.quantity;
            }
        }
    }
}